### PR TITLE
Add API for counting agents 

### PR DESCRIPTION
**Pull request by mgrachev**

### CODE REVIEW

File: src/db/agent_session.rs [hunk 1]
```diff
@@ -34,3 +35,39 @@ impl AgentList {
         .await
     }
 }
+
+#[derive(Serialize)]
+pub struct AgentCount {
+    pub classroom_id: ClassroomId,
+    pub count: Option<i64>,
```
Comment (khodzha) [inline src/db/agent_session.rs:42]: why option?

File: src/app/api/v1/counter/mod.rs [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+    let result = AgentCounter::new(payload.classroom_ids)
+        .execute(&mut conn)
+        .await
+        .context("Failed to count agents")
+        .error(ErrorKind::DbQueryFailed)?;
+
+    let mut agents_count = HashMap::new();
+    result.iter().for_each(|a| {
+        agents_count.insert(a.classroom_id, a.count.unwrap_or_default());
```
Comment (khodzha) [inline src/app/api/v1/counter/mod.rs:45]: you can probably use [FromIterator](https://doc.rust-lang.org/std/iter/trait.FromIterator.html) and [collect](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.collect) here

something like
```
result.iter().map(|a| (a.classroom_id, a.count)).collect::<HashMap<_, _>>()
```

File: src/db/agent_session.rs [hunk 3]
```diff
@@ -34,3 +35,39 @@ impl AgentList {
...
+    // TODO: Use Vec<ClassroomId> instead
+    classroom_ids: Vec<Uuid>,
+}
+
+impl AgentCounter {
+    pub fn new(classroom_ids: Vec<Uuid>) -> Self {
+        Self { classroom_ids }
+    }
+
+    pub async fn execute(&self, conn: &mut PgConnection) -> sqlx::Result<Vec<AgentCount>> {
```
Comment (khodzha) [inline src/db/agent_session.rs:55]: i think returning HashMap instead of Vec here better explains query intent and underscores classroom_id uniqueness

**Review by khodzha — APPROVED** [review]

**Review by manifest — APPROVED** [review]

File: src/app/router.rs [hunk 4]
```diff
@@ -22,7 +22,11 @@ fn api_router() -> Router {
         .metered_route("/api/v1/healthz", get(v1::healthz))
         .metered_route(
             "/api/v1/classrooms/:classroom_id/agents",
-            options(v1::options).get(v1::classroom::list_agents::<AppState>),
+            get(v1::classroom::list_agents::<AppState>).options(v1::options),
+        )
+        .metered_route(
+            "/api/v1/counters/agent",
+            post(v1::counter::count_agents::<AppState>).options(v1::options),
```
Comment (khodzha) [inline src/app/router.rs:29]: we dont need options handler here, this is internal api

**Review by khodzha — APPROVED** [review]

**Pull request merged**